src/core_editor/line_buffer.rs

```diff
@@ -165,21 +160,20 @@ impl LineBuffer {
     /// TODO: Check unicode validation
     pub fn insert_str(&mut self, string: &str) {
         let pos = self.insertion_point();
-        self.lines[pos.line].insert_str(pos.offset, string);
+        self.lines.insert_str(pos.offset, string);
         self.insertion_point.offset = pos.offset + string.len();
     }
 
     /// Empty buffer and reset cursor
     pub fn clear(&mut self) {
-        self.lines.clear();
-        self.lines.push(String::new());
+        self.lines = String::new();
         self.insertion_point = InsertionPoint::new();
     }
 
     /// Clear everything beginning at the cursor to the right/end.
     /// Keeps the cursor at the end.
     pub fn clear_to_end(&mut self) {
-        self.lines[self.insertion_point.line].truncate(self.insertion_point.offset);
+        self.lines.truncate(self.insertion_point.offset);
```

Comment:
Won't this delete the whole thing? Can we add a test for this one?

Reply:
Not sure what you mean here